src/transpilation/macro.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceNode } from "source-map";
+import { escapeString, unescapeLuaString } from "../LuaPrinter";
+
+export type ResolveMacroReplacer = (request: string) => string | { error: string };
+
+export function replaceResolveMacroSourceNodes(rootNode: SourceNode, replacer: ResolveMacroReplacer) {
+    function walkSourceNode(node: SourceNode, parent: SourceNode) {
+        for (const child of node.children) {
+            if ((child as any) === "__TS__Resolve") {
```

Comment:
This file name 'macro' is confusing imo. Would bit be clearer to rename to `resolve` or `resolve-macro`?